octopus/src/main/scala/octopus/App.scala

```diff
@@ -0,0 +1,33 @@
+package octopus
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait App[M[_]] {
+  def pure[A](a: A): M[A]
+  def map2[A, B, C](first: M[A], second: M[B])(combine: (A, B) => C): M[C]
+  def recover[A, B <: A](app: M[A], f: Throwable => B): M[A]
+  def map[A, B](fa: M[A])(f: A => B): M[B] = {
+    map2(fa, pure(true))((a, _) => f(a))
+  }
+}
+
+object App extends LowPriorityAppImplicits {
+  def apply[M[_]](implicit a: App[M]): App[M] = a
+}
+
+trait LowPriorityAppImplicits {
+
+  implicit def futureApp(implicit ec: ExecutionContext): App[Future] = new App[Future] {
```

Comment:
you can override `map` here so that `Future.map` is used instead of default, inefficient implementation

Reply:
map implementation now looks like this:
`override def map[A, B](fa: Future[A])(f: A => B): Future[B] = fa.map(f)`